src/search.ml

```diff
@@ -83,39 +83,37 @@ module Make
   let look_around array key key_metric index =
     let rec search (op : int64 -> int64) curr =
       let i = op curr in
-      if i < 0L || i >= Array.length array then None
+      if i < 0L || i >= Array.length array then raise Not_found
       else
         let e = array.(i) in
         let e_metric = Metric.of_entry e in
-        if not Metric.(key_metric = e_metric) then None
-        else if Key.equal (Entry.to_key e) key then Some (Entry.to_value e)
+        if not Metric.(key_metric = e_metric) then raise Not_found
+        else if Key.equal (Entry.to_key e) key then Entry.to_value e
         else (search [@tailcall]) op i
     in
-    match search Int64.succ index with
-    | Some e -> Some e
-    | None -> (search [@tailcall]) Int64.pred index
+    (search [@tailcall]) Int64.pred index
```

Comment:
This now only searches backwards: have we changed something that makes searching forwards unnecessary? If so, I think it's probably worth renaming `look_around` to reflect that it's only looking in one direction.

Reply:
Oops my bad, good catch!

Reply:
As an aside, we should have tests that expose this behaviour using a degenerate hash function.